contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -191,51 +134,130 @@ contract MixinStakingPoolRewards is
-    /// @dev Computes a member's reward over a given epoch interval.
-    /// @param poolId Uniqud Id of pool.
-    /// @param memberStakeOverInterval Stake delegated to pool by meber over the interval.
-    /// @param beginEpoch beginning of interval.
-    /// @param endEpoch end of interval.
-    /// @return rewards accumulated over interval [beginEpoch, endEpoch]
-    function _computeMemberRewardOverInterval(
+    /// @dev Transfers a delegators accumulated rewards from the transient pool Reward Pool vault
+    ///      to the Eth Vault. This is required before the member's stake in the pool can be
+    ///      modified.
+    /// @param poolId Unique id of pool.
+    /// @param member The member of the pool.
+    function _transferDelegatorRewardsToEthVault(
         bytes32 poolId,
-        uint256 memberStakeOverInterval,
-        uint256 beginEpoch,
-        uint256 endEpoch
+        address member,
+        IStructs.StoredBalance memory unsyncedDelegatedStakeToPoolByOwner,
+        uint256 currentEpoch
     )
         private
-        view
-        returns (uint256)
     {
-        IStructs.Fraction memory beginRatio = cumulativeRewardsByPool[poolId][beginEpoch];
-        IStructs.Fraction memory endRatio = cumulativeRewardsByPool[poolId][endEpoch];
-        uint256 reward = LibFractions.scaleFractionalDifference(
-            endRatio.numerator,
-            endRatio.denominator,
-            beginRatio.numerator,
-            beginRatio.denominator,
-            memberStakeOverInterval
+        // compute balance owed to delegator
+        uint256 balance = _computeRewardBalanceOfDelegator(
+            poolId,
+            unsyncedDelegatedStakeToPoolByOwner,
+            currentEpoch
         );
-        return reward;
+        if (balance == 0) {
+            return;
+        }
+
+        // transfer from transient Reward Pool vault to ETH Vault
+        _transferMemberBalanceToEthVault(poolId, member, balance);
+    }
+
+    /// @dev Computes the reward balance in ETH of a specific member of a pool.
+    /// @param poolId Unique id of pool.
+    /// @param unsyncedDelegatedStakeToPoolByOwner Unsynced delegated stake to pool by owner
+    /// @param currentEpoch The epoch in which this call is executing
+    /// @return totalReward Balance in ETH.
+    function _computeRewardBalanceOfDelegator(
+        bytes32 poolId,
+        IStructs.StoredBalance memory unsyncedDelegatedStakeToPoolByOwner,
+        uint256 currentEpoch
+    )
+        private
+        view
+        returns (uint256 totalReward)
+    {
+        // value is always zero in these two scenarios:
+        //   1. The owner's delegated is current as of this epoch: their rewards have been moved to the ETH vault.
+        //   2. The current epoch is zero: delegation begins at epoch 1
+        if (unsyncedDelegatedStakeToPoolByOwner.currentEpoch == currentEpoch || currentEpoch == 0) return 0;
+
+        // compute reward accumulated during `delegatedStake.currentEpoch`;
```

Comment:
Really clean implementation! The `computeMemberRewardOverInterval` function is really nice here.